api/v1/storage-contract.go

```diff
@@ -34,22 +42,56 @@ type AddressedStorageContract struct {
 //	@Description <br/> *Note* CID is optional, if specified, then `storage_contract` becomes optional.
 //	@Description If both are specified, then we will validate that the CID matches the proposed storage contract and return an error if not.
 //	@Description If only CID is specified, then we will fetch it and update the storage contract to it.
-//	@Security apiKey
+//	@Security		apiKey
 //	@Param 			collection body AddressedStorageContract true "New Storage Contract to update to"
 //	@Produce		json
 //	@Success		200	{object}	ResponseEnvelope{response=AddressedStorageContract}
 //	@Router			/storage-contract [post]
-func handleSetStorageContract(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+func (a *apiV1) handleSetStorageContract(c echo.Context) error {
+	var addressedStorageContract AddressedStorageContract
+	err := json.NewDecoder(c.Request().Body).Decode(&addressedStorageContract)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	if len(addressedStorageContract.Cid) == 0 {
```

Comment:
we should also validate that it parses to a cid i.e https://github.com/ipfs/go-cid?tab=readme-ov-file#parsing-string-input-from-users